http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcClientConfigurationBuilder.java

```diff
@@ -100,6 +102,32 @@ protected OidcClientConfiguration internalBuild(final OidcJsonConfiguration oidc
         if (oidcJsonConfiguration.getTokenCookiePath() != null) {
             oidcClientConfiguration.setOidcStateCookiePath(oidcJsonConfiguration.getTokenCookiePath());
         }
+        if (oidcJsonConfiguration.getAuthenticationRequestFormat() != null) {
+            oidcClientConfiguration.setAuthenticationRequestFormat(oidcJsonConfiguration.getAuthenticationRequestFormat());
+        } else {
+            oidcClientConfiguration.setAuthenticationRequestFormat(REQUEST_TYPE_OAUTH2.getValue());
+        }
+        if (oidcJsonConfiguration.getRequestSignatureAlgorithm() != null) {
+            oidcClientConfiguration.setRequestSignatureAlgorithm(oidcJsonConfiguration.getRequestSignatureAlgorithm());
+        } else {
+            oidcClientConfiguration.setRequestSignatureAlgorithm(NONE);
+        }
+        if (oidcJsonConfiguration.getRequestEncryptAlgorithm() != null && oidcJsonConfiguration.getRequestContentEncryptionMethod() != null) { //both are required to encrypt the request object
```

Comment:
Might be good to throw an IllegalArgumentException if only is specified if that's invalid configuration.

Reply:
Since we know that we'll need to encrypt the request object, maybe we can set the public key locator for that here on the `oidcClientConfiguration` (e.g., `oidcClientConfiguration.setEncPublicKeyLocator` or something like that) similar to the way we set the public key locator for the SIG case.